src/seedu/addressbook/AddressBook.java

```diff
@@ -986,18 +1004,31 @@ private static String extractNameFromPersonString(String encoded) {
     private static String extractPhoneFromPersonString(String encoded) {
         final int indexOfPhonePrefix = encoded.indexOf(PERSON_DATA_PREFIX_PHONE);
         final int indexOfEmailPrefix = encoded.indexOf(PERSON_DATA_PREFIX_EMAIL);
+        final int indexOfBirthdatePrefix = encoded.indexOf(PERSON_DATA_PREFIX_BIRTHDATE);
 
         // phone is last arg, target is from prefix to end of string
-        if (indexOfPhonePrefix > indexOfEmailPrefix) {
+        if ((indexOfPhonePrefix > indexOfEmailPrefix)&&(indexOfPhonePrefix > indexOfBirthdatePrefix)) {
```

Comment:
Note that there should be a space after operators.
Do refer to the [coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html)